python/lsst/ts/envsensors/device/mock_device.py

```diff
@@ -106,31 +96,105 @@ def _format_temperature(self, i: int):
         s: `str`
             A string representing a temperature.
         """
-        temp = random.uniform(MockTemperatureConfig.min, MockTemperatureConfig.max)
         if i < self._missed_channels:
             return ""
+
+        prefix = f"C{i:02d}="
+        value = random.uniform(MockTemperatureConfig.min, MockTemperatureConfig.max)
         if i == self._disconnected_channel:
-            return f"C{i:02d}={DISCONNECTED_VALUE}"
-        return f"C{i:02d}={temp:09.4f}"
+            value = float(DISCONNECTED_VALUE)
+        return f"{prefix}{value:09.4f}"
+
+    def _format_hx85a(self, i: int):
+        """Creates a formatted string representing a temperature for the given
+        channel. All channels for this sensor are always connected.
+
+        Parameters
+        ----------
+        i: `int`
+            The 0-based temperature channel.
+        Returns
+        -------
+        s: `str`
+            A string representing a temperature.
+        """
+        if i < self._missed_channels:
+            return ""
+
+        prefix = ""
+        value = 0.0
+        if i == 0:
+            prefix = "%RH="
+            value = random.uniform(MockHumidityConfig.min, MockHumidityConfig.max)
+            return f"{prefix}{value:5.2f}"
```

Comment:
You could reduce duplication by having a method for each of the 4 kinds of values the hx85 sensors can return, then call the method based on the index.

Reply:
Agreed. I split up the two functions into 4 new ones that together take care of each possible value for the two sensors and removed these two functions.